cmd/util/ledger/util/payload_grouping_test.go

```diff
@@ -0,0 +1,153 @@
+package util_test
+
+import (
+	"crypto/rand"
+	"encoding/hex"
+	rand2 "math/rand"
+	"runtime"
+	"testing"
+
+	"github.com/rs/zerolog"
+	"github.com/stretchr/testify/require"
+
+	"github.com/onflow/cadence/runtime/common"
+
+	"github.com/onflow/flow-go/cmd/util/ledger/util"
+	"github.com/onflow/flow-go/ledger"
+	"github.com/onflow/flow-go/ledger/common/convert"
+	"github.com/onflow/flow-go/model/flow"
+)
+
+func TestGroupPayloadsByAccount(t *testing.T) {
+	log := zerolog.New(zerolog.NewTestWriter(t))
+	payloads := generateRandomPayloads(1000000)
+	tmp := make([]*ledger.Payload, len(payloads))
+	copy(tmp, payloads)
+
+	groups := util.GroupPayloadsByAccount(log, payloads, 0)
+
+	require.Greater(t, groups.Len(), 1)
+}
+
+func TestGroupPayloadsByAccountCompareResults(t *testing.T) {
+	log := zerolog.Nop()
+	payloads := generateRandomPayloads(1000000)
+	tmp1 := make([]*ledger.Payload, len(payloads))
+	tmp2 := make([]*ledger.Payload, len(payloads))
+	copy(tmp1, payloads)
+	copy(tmp2, payloads)
+
+	groups1 := util.GroupPayloadsByAccount(log, tmp1, 0)
+	groups2 := util.GroupPayloadsByAccount(log, tmp2, runtime.NumCPU())
+
+	groups3 := map[common.Address][]*ledger.Payload{}
```

Comment:
For the payloads from the same address, we only compared the length, it's better to check the payloads in the same group are the same. We could do that by turning the payload group from a slice to a map, and check each of the payload in the same group exists in the map.
```suggestion
	groups3 := map[common.Address]map[ledger.Path]*ledger.Payload
```

Reply:
I added a `require.ElementsMatch` to check for this